app/controllers/users_controller.rb

```diff
@@ -20,9 +20,23 @@ def destroy
     redirect_to users_url
   end
 
-  def create
-    @user = User.new user_params
-    attempt_save
+  def import
+    if request.post?
+      xml = Nokogiri::XML params.require(:file).open
+      statuses = User.import_from_xml(xml)
+      if statuses
+        message = "Imported #{statuses[:imported]} new users"
+        unless statuses[:updated].zero?
+          message += " and updated #{statuses[:updated]}"
+        end
+        message += '.'
+        unless statuses[:rejected].zero?
+          message << " #{statuses[:rejected]} were rejected."
+        end
+        redirect_to users_url, notice: message
```

Comment:
Hmm, good point, I suppose if they're all rejected we should just call that an outright failure and it should be an `alert`. I guess the question is what to do with falling somewhere in between — some mixture of acceptance and rejection. Maybe an alert if *any* are rejected?